corehq/apps/integration/kyc/services.py

```diff
@@ -0,0 +1,39 @@
+from corehq.apps.integration.kyc.models import UserDataStore
+from corehq.form_processor.models import CommCareCase
+
+
+class CustomUserDataNotFound(Exception):
+    pass
+
+
+def get_user_data_for_api(user, config):
+    custom_user_data = _get_custom_user_data(user, config)
+    user_data_for_api = {}
+    for mapping in config.api_field_to_user_data_map:
+        try:
+            if mapping['source'] == 'standard':
+                user_data_for_api[mapping['fieldName']] = getattr(user, mapping['mapsTo'])
+            else:
+                user_data_for_api[mapping['fieldName']] = custom_user_data[mapping['mapsTo']]
+        except (AttributeError, KeyError):
+            # Conservative approach to skip the API field if data is not available for the user
+            continue
+    return user_data_for_api
+
+
+def _get_custom_user_data(user, config):
```

Comment:
I'm concerned that "custom_user_data" here is ambiguous. It could refer generally to data to be sent to the KYC API, or it could refer specifically to the custom user data returned by  `CommCareUser.get_user_data()`.

The other thing is that the reason that some projects are using a custom case type is because KYC is being used for non-users. They could be beneficiaries, or other people involved in the project whose identities need to be confirmed. Maybe rename to something like "_get_source_data()"?

```suggestion
get _get_source_data(source, config):
    """
    Returns a dictionary of source data.

    ``source`` is a CommCareUser or a CommCareCase.
    """
```

Reply:
Thanks Norman for pointing this out. I was not aware earlier that KYC can be used for no-users. 
The suggestion sounds good.  Also made change so that source can be Commcare Case for custom case type.

Addressed in https://github.com/dimagi/commcare-hq/pull/35690/commits/5e32f5bb0d7bc9e61ec2d1317449999fb230c465